cheroot/wsgi.py

```diff
@@ -276,6 +278,11 @@ def get_environ(self):
 
         return env
 
+    def _get_peer_uid(self):
+        creds = self.req.server.socket.getsockopt(socket.SOL_SOCKET, socket.SO_PEERCRED, struct.calcsize('3i'))
```

Comment:
I felt this was a lot more connected to REMOTE_ADDR and the likes, as I've felt it is a property of the connection rather than the request itself.

In our internal HTTP based service, we are using the environment to populate our internal data structures and by that time the connection itself is not available.

If you meant that _REMOTE_UID stays in the environment, but the implementation itself moves to HTTPConnection, that could make sense. Is that what you are thinking?

Thanks for the feedback.